src/v/cloud_storage/recovery_request.cc

```diff
@@ -14,6 +14,10 @@
 #include "json/validator.h"
 #include "json/writer.h"
 
+#include <seastar/core/coroutine.hh>
+#include <seastar/util/short_streams.hh>
+
+namespace cloud_storage {
```

Comment:
Question: should the anonymous namespace be inside the `cloud_storage` namespace? It doesn't seem like this will cause any problems in terms of visibility, but is this intentionally nested?

Reply:
I moved it so that validate_headers has access to bad_request without fully specifying the namespace.

This is generally a common pattern I use to hide visibility instead of the static keyword, it's nice to wrap in your namespace so that you don't have to start qualifying everything. Related is this clang tidy check: https://releases.llvm.org/16.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/misc/use-anonymous-namespace.html